vda5050_connector/vda5050_connector_py/vda5050_controller.py

```diff
@@ -1032,7 +1032,10 @@ def _match_stitch_nodes(self, order: VDAOrder):
             True if last <> first base nodes match, False otherwise.
 
         """
-        last_node = self._current_order.nodes[-1]
+        base_order_nodes = [
+            node for node in self._current_order.nodes if node.released
+        ]
+        last_node = base_order_nodes[-1]
```

Comment:
Is it guaranteed that `self._current_order.nodes` would have at least 1 node? I think it might be a good idea to prevent `IndexError` JIC.

Other than that it looks good to me. Thanks!